### DISCUSSION TITLE
Cron jobs saved as binary aren't migrated to JSON

### DISCUSSION THREAD

**Discussion by iuliusandu75** [quartznet/quartznet#2457] [Q&A]
Hey everyone,

I wonder if anyone has come across this.
We have a product using Quartz which we want to migrate to .net core 8 and run in containers. 
We started with the solution build in net framework and using binary serialisation. Obviously that is not supported in .net8 and we started the process of migrating to JSON serialisation. That's all fine and we were able to let all existing one-off jobs (serialised as binary) execute and now new jobs are stored as JSON. We used a custom migrator serialiser for that which we found in the documentation.

The issue is that legacy cron jobs are not updated to JSON when they get executed, say we have a job that runs every 10 min, that will stay as binary in the DB ([dbo].[QRTZ_JOB_DETAILS]). How we can we migrate these to JSON format so we can stop using binary serialisation in the code. If that's not possible, can we delete the details for these jobs in a safe way so we can recreate them using JSON format.

![quartz](https://github.com/quartznet/quartznet/assets/91681508/594917c0-5d9e-4b8c-9297-35a9af67b480)

Thanks,
Iuliu

**Comment by lahma** [comment]
Hmm. I can see cases where data wouldn't get updated, I think this needs some fine-tuning and update to guidance (migrator sample)..

**Reply by iuliusandu75** [reply]
Thanks Marko. 
I was peeking at the code inside Quartz.NET and noticed an attribute called PersistJobDataAfterExecutionAttribute which could trigger an update to the job data after execution which I think could fix the issue in a sense that legacy binary jobs would fix themselves given enough time. Do you think decorating my jobs with attribute (presumably I also need DisallowConcurrentExecution) could work?

**Reply by lahma** [reply]
I'll update the sample soon and let's see if it's of any help..

**Reply by lahma** [reply]
Here's update to migration serializer example which explicitly ensure that job's data will be marked dirty and thus being candidate to serialization even if job itself doesn't update the data.

https://github.com/quartznet/quartznet/pull/2458/files#diff-e346bf18ef9a9b01d4147a950e5b763663ace548bf3daac66eacf6caf79e6d1dR87-R99

Could this have been the source of problem?
